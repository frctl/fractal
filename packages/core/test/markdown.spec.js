const md = require('../src/markdown');

const code = `
\`\`\`
module.exports = {
    foo: 'bar',
};
\`\`\`
`;

const codeWithLang = `
\`\`\`js
module.exports = {
    foo: 'bar',
};
\`\`\`
`;

describe('Markdown renderer', () => {
    it('does not directly mutate the supplied configuration object', () => {
        const config = {};
        md('**foo**', config);
        expect(config).toEqual({});
    });

    it('renders markdown correctly', () => {
        const result = md('**foo**');
        expect(result).toMatchSnapshot();
    });

    it('renders code correctly', () => {
        const result = md(code);
        expect(result).toMatchSnapshot();
    });

    it('renders code with lang correctly', () => {
        const result = md(codeWithLang);
        expect(result).toMatchSnapshot();
    });
});
