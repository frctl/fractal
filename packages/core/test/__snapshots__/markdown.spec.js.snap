// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renderer renders code correctly 1`] = `
"<pre><code class=\\"hljs\\"><span class=\\"hljs-built_in\\">module</span>.<span class=\\"hljs-built_in\\">exports</span> = {
    foo: <span class=\\"hljs-string\\">&#x27;bar&#x27;</span>,
};</code></pre>"
`;

exports[`Markdown renderer renders code with lang correctly 1`] = `
"<pre><code class=\\"hljs language-js\\"><span class=\\"hljs-built_in\\">module</span>.exports = {
    <span class=\\"hljs-attr\\">foo</span>: <span class=\\"hljs-string\\">&#x27;bar&#x27;</span>,
};</code></pre>"
`;

exports[`Markdown renderer renders markdown correctly 1`] = `
"<p><strong>foo</strong></p>
"
`;
